src/i_poll_events.hpp

```diff
@@ -30,20 +30,30 @@
 #ifndef __ZMQ_I_POLL_EVENTS_HPP_INCLUDED__
 #define __ZMQ_I_POLL_EVENTS_HPP_INCLUDED__
 
+#include "fd.hpp"
+
 namespace zmq
 {
 // Virtual interface to be exposed by object that want to be notified
 // about events on file descriptors.
 
 struct i_poll_events
 {
+    typedef void * handle_t;
```

Comment:
if anything, this should probably be a typedef based on poller_t::handle_t (which needs to be moved somewhere else since this introduces an implicit cyclic dependency between i_poll_events and poller_t)

Reply:
I'm totally agree with you. I can put the handle definition in another file it will be nicer